### PR TITLE
Pin port

### PR DESCRIPTION
**Pull request by ghost**
Ports now are pinnable/unpinnable through the UI. The unpin button shows up when the port is pinned (dragging a port pins it), it is hidden unless the user hovers over the port element or the pin element.

image

### CODE REVIEW

File: lib/viewcomponents/vertex.jsx [hunk 1]
```diff
@@ -155,7 +155,12 @@ var Vertex = React.createClass({
     var vPct = pos.y/styles.get('height');
 
     var cardinalPosition = PositionUtils.Conversion.cartesianToCardinal({x: hPct, y: vPct});
-
+    /**this.portModel.setAttrs({
+        isPinned: true,
+		position: cardinalPosition
+      });
+	  */
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: Remove this.

File: lib/viewcomponents/vertex.jsx [hunk 2]
```diff
@@ -155,7 +155,12 @@ var Vertex = React.createClass({
...
+	portModel.set('isPinned',true);
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: indentation and spacing

File: lib/viewcomponents/vertex.jsx [hunk 3]
```diff
@@ -155,7 +155,12 @@ var Vertex = React.createClass({
...
     portModel.set('position', cardinalPosition);
```
Comment (jboolean) [inline lib/viewcomponents/vertex.jsx]: Can you combine these sets into a setAttrs?

File: lib/viewcomponents/vertex.jsx [hunk 4]
```diff
@@ -155,8 +155,10 @@ var Vertex = React.createClass({
     var vPct = pos.y/styles.get('height');
 
     var cardinalPosition = PositionUtils.Conversion.cartesianToCardinal({x: hPct, y: vPct});
-
-    portModel.set('position', cardinalPosition);
+    portModel.setAttrs({
+        isPinned: true,
+		position: cardinalPosition
+      });
```
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: These three lines are improperly indented.

Comment (salva-abate) [inline lib/viewcomponents/vertex.jsx]: That looks super janky and not at all how it looks on my machine lol but I'll take a look maybe its the editor

**Comment by jboolean** [comment]
**Approve**

**Pull request merged**